examples/dynamic-lora-sidecar/sidecar/sidecar.py

```diff
@@ -0,0 +1,204 @@
+import requests
+import yaml
+import time
+from watchdog.observers import Observer
+from watchdog.events import FileSystemEventHandler
+import logging
+import datetime
+import os
+
+CONFIG_MAP_FILE = os.environ.get("DYNAMIC_LORA_ROLLOUT_CONFIG", "configmap.yaml")
+DYNAMIC_LORA_FLAG = "VLLM_ALLOW_RUNTIME_LORA_UPDATING"
+BASE_FIELD = "vLLMLoRAConfig"
+logging.basicConfig(
+    level=logging.INFO, format="%(asctime)s - %(levelname)s - %(message)s"
+)
+
+
+def current_time_human() -> str:
+    now = datetime.datetime.now(datetime.timezone.utc).astimezone()
+    return now.strftime("%Y-%m-%d %H:%M:%S %Z%z")
+
+
+class ConfigWatcher(FileSystemEventHandler):
+    def __init__(self, callback):
+        """
+        Watches config
+
+        Args:
+            callback : Callback function taking no parameters and no return values parsed
+        """
+        self.callback = callback
+
+    def on_modified(self, event):
+        if not event.is_directory and event.src_path.endswith(".yaml"):
+            logging.info(
+                f"Config '{event.src_path}' modified! at '{current_time_human()}'"
+            )
+            self.callback()
+
+
+class LoraReconciler:
+    """
+    Reconciles adapters registered on vllm server with adapters listed in configmap in current state
+    """
+
+    def __init__(self):
+        self.deployment_name = ""
+        self.registered_adapters = {}
+        self.config_map_adapters = {}
+        self.health_check_timeout = datetime.timedelta(seconds=150)
+        self.health_check_interval = datetime.timedelta(seconds=15)
+        if not self.validate_dynamic_lora():
+            logging.fatal(f"{DYNAMIC_LORA_FLAG} set to False")
+        self.load_configmap()
+        self.get_registered_adapters()
+
+    def validate_dynamic_lora(self):
+        """Validate if dynamic lora updating is enabled"""
+        return os.environ.get(DYNAMIC_LORA_FLAG, False)
+
+    def load_configmap(self):
+        """Load configmap into memory"""
+        with open(CONFIG_MAP_FILE, "r") as f:
+            deployment = yaml.safe_load(f)[BASE_FIELD]
```

Comment:
Let's also do a pass and rename "deployments" in this file

Reply:
Done